packages/react/src/components/slot/index.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { MaybeArray } from "../../domain/types";
+
+export type Props<Name extends string> = {
+  children: React.ReactNode;
+  name: Name; // name of the slot the component is being rendered into
+};
+
+export type Slots<SlotNames extends string> = MaybeArray<
```

Comment:
Unfortunately at the moment TypeScript cannot test the children types:
- https://www.arahansen.com/how-children-types-work-in-react-18-and-typescript-4/
- https://stackoverflow.com/a/71987357
- https://github.com/microsoft/TypeScript/issues/21699

I still wanted to add this type in case it changes in the future - for the moment we can at least use it as docs, even though it is not enforced.

Reply:
I think we could enforce the type on the name at least with this suggestion: https://github.com/slashid/javascript/pull/155#discussion_r1325874300